src/pages/EthRealTimeLockPage.tsx

```diff
@@ -218,39 +212,22 @@ const EthRealTimeLockPage: React.FC<Props> = ({ lockdropNetwork }) => {
 
     // refresh if contract reloads
     useEffect(() => {
-        if (web3) {
-            setLoading({
-                loading: true,
-                message: 'Connecting to Web3 instance with new contract...',
-            });
-
-            (async function() {
-                // fetch a new contract
-                const _contract = await ethLockdrop.createContractInstance(web3, contractAddress);
-                // fetch new lock events
-                await handleFetchLockEvents(_contract);
-                // check contract start and end dates
-                const _end = await ethLockdrop.getContractEndDate(_contract);
-                const _start = await ethLockdrop.getContractStartDate(_contract);
-                setLockdropEnd(_end);
-                setLockdropStart(_start);
-                setContract(_contract);
-            })()
-                .catch(e => {
-                    console.log(e);
-                    toast.error(e.message);
-                })
-                .finally(() => {
-                    setLoading({ loading: false, message: '' });
+        if (isChangingContract) {
+            if (!isWeb3Loading) {
+                setLoading({
```

Comment:
Since the page loading screen is used for multiple pages, it could be better to create a loading overlay as a separate component so we can handle error behaviors and messages from a single location.
For example, like this:
- https://github.com/polkadot-js/apps/blob/e202dda584900a73dc7cd565fbc929cb02e38d02/packages/apps/src/overlays/Connecting.tsx#L21
- https://github.com/polkadot-js/apps/blob/ccb488626e3ee8e445fd6157a2fe0232fa40ce30/packages/apps/src/Apps.tsx#L42